### ISSUE TITLE
[question] Implementing Learning Rate Function in PPO2

### ISSUE THREAD

**Issue by fcossio** [hill-a/stable-baselines#255]
Hi, I'm implementing PPO2 and I would like to use a function for the learning_rate parameter, How can I pass a function at the creation of the model like this:
```
def learning_rate( alpha_zero, decay_rate, timestep ):
   learning_rate = alpha_zero/(1+decay_rate * timestep)
   return learning_rate
```
The question I have regards on how to use the "actual timestep" parameter when model.learn() is used.
I would like to create a model like:
`model = PPO2(MlpPolicy, env, learning_rate = learning_rate(alpha_zero, decay_rate))`

I'm sorry it might be a noob question but I couldn't find any examples of a function or schedule to change the learning rate

**Comment by araffin** [comment]
Hello,

what you are looking for is called `functools.partial` in python ;)

See [documentation](https://docs.python.org/3.6/library/functools.html#functools.partial)

**Comment by araffin** [comment]
An example:

```python
def linear_schedule(initial_value):
    """
    Linear learning rate schedule.
    :param initial_value: (float or str)
    :return: (function)
    """
    if isinstance(initial_value, str):
        initial_value = float(initial_value)

    def func(progress):
        """
        Progress will decrease from 1 (beginning) to 0
        :param progress: (float)
        :return: (float)
        """
        return progress * initial_value

# schedule is a callable
schedule = linear_schedule(0.1)

assert schedule(1) == 0.1
```
return func

**Comment by fcossio** [comment]
Thank you very much for the fast response!
I was able to implement it like this:
```
def lr_func(alpha_zero, decay_rate):
    def lr(timestep):
        learning_rate = alpha_zero * 1/( 1 + decay_rate * timestep)
        return learning_rate
    return lr
```
and then editing line [289](https://github.com/hill-a/stable-baselines/blob/c3332711e91afe0f0d497efef17e4c3f6865755f/stable_baselines/ppo2/ppo2.py#L289) to 
`lr_now = self.learning_rate(self.timestep)`
